projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -274,6 +261,41 @@ export class DatagridComponent<R> implements OnInit {
     @Input() trackBy: TrackByFunction<R> = (index: number, record: (R & HasHref) | undefined): string | number =>
         record && (record.href || index)
 
+    /**
+     * Returns the buttons that should be featured given the {@param selection} of currently selected entities.
+     */
+    getFeaturedButtons(selection: R[]): ContextualButton<R>[] {
+        let toTake = this.buttonConfig.contextualButtons.featuredCount;
+        const toOutput = [];
+        this.buttonConfig.contextualButtons.featured.forEach(featured => {
+            if (toTake === 0) {
+                return;
+            }
+            const buttons = this.buttonConfig.contextualButtons.buttons.filter(button => button.id === featured);
+            if (buttons.length === 1) {
+                const button = buttons[0];
+                if (button.shouldDisplay(selection)) {
+                    toOutput.push(button);
+                    toTake -= 1;
```

Comment:
Instead of modifying `toTake`, can't we just always compare the length of `toOutput`?